views/js/runner/plugins/controls/progressbar/progress.js

```diff
@@ -252,12 +307,36 @@ define([
                 if (section.stats.questions > 0) {
                     updateDetails(stats.answerableSections, section, currentItem.position);
                 }
+
+                if (config.indicator === 'categories') {
+                    _.forEach(section.items, function (item) {
+                        if (matchCategories(item.categories, categoriesToMatch)) {
+                            updateItemDetails(stats.matchedCategories, item, currentItem.position);
+                        }
+                    });
+                }
             });
         });
 
         return stats;
     }
 
+    function matchCategories(categories, expectedCategories) {
+        var categoryId;
+
+        if (expectedCategories && expectedCategories.length) {
+            // all expected categories have to be included in the provided categories
+            for (categoryId in expectedCategories) {
+                if (expectedCategories.hasOwnProperty(categoryId)
+                    && categories.indexOf(expectedCategories[categoryId]) === -1) {
+                    return false;
+                }
+            }
+        }
```

Comment:
The main algo is ok, but you could also improve it by using a hashtable instead of looping over the categories.

For instance, before to start looping over the testMap (line 290), you could transform the list of expected categories into a hashtable:
```javascript
var matchSize = config.categories && config.categories.length;
var categoriesToMatch = matchSize && _.reduce(config.categories, function(map, category) {
    map[category] = true;
    return map;
}, {});
```

And then to check it (line 311):
```
                if (config.indicator === 'categories') {
                    _.forEach(section.items, function (item) {
                        if (matchCategories(item.categories, categoriesToMatch, matchSize) {
                            updateItemDetails(stats.matchedCategories, item, currentItem.position);
                        }
                    });
                }
```

And the match function
```
/**
 * 
 * @param {Array} categories - List of categories to check
 * @param {Object} expectedCategories - Hashtable of expected categories
 * @param {Number} minWanted - Minimal number of expected categories that should match
 * @returns {Boolean}
 */
function matchCategories(categories, expectedCategories, minWanted) {
    var matched = 0;
    
    if (expectedCategories) {
        _.forEach(categories, function(category) {
            if (expectedCategories[category]) {
                matched ++;
            }
        });
    }
    return matched === minWanted;
}
```